packages/fuels-programs/src/calls/contract_call.rs

```diff
@@ -0,0 +1,611 @@
+use std::{collections::HashMap, fmt::Debug, marker::PhantomData, panic};
+
+use fuel_tx::{Address, AssetId, Output, Receipt, Transaction};
+use fuel_vm::fuel_asm::PanicReason;
+use fuels_core::{
+    abi_encoder::UnresolvedBytes,
+    constants::FAILED_TRANSFER_TO_ADDRESS_SIGNAL,
+    offsets::call_script_data_offset,
+    parameters::{CallParameters, TxParameters},
+    traits::Tokenizable,
+};
+use fuels_signers::{
+    provider::{Provider, TransactionCost},
+    Signer, WalletUnlocked,
+};
+use fuels_types::{
+    bech32::Bech32ContractId,
+    core::{Selector, Token},
+    errors::Error,
+    param_types::ParamType,
+};
+
+use crate::calls::contract_call_utils::{
+    build_script_data_from_contract_calls, calculate_required_asset_amounts, get_instructions,
+    get_single_call_instructions, get_transaction_inputs_outputs, CallOpcodeParamsOffset,
+};
+use crate::{
+    calls::call_response::FuelCallResponse,
+    calls::call_utils::get_decoded_output,
+    execution_script::ExecutableFuelCall,
+    logs::{decode_revert_error, LogDecoder},
+};
+
+/// How many times to attempt to resolve missing tx dependencies.
+pub const DEFAULT_TX_DEP_ESTIMATION_ATTEMPTS: u64 = 10;
```

Comment:
A duplicate can be found in `fuels_programs::contract`. Should probably be deleted since it isn't used.